openspec/changes/set-env-java-home-from-java-version/design.md

```
@@ -0,0 +1,231 @@
+## Context
+
+- `TeamcityCreateBuildChainCommand.createBuildChain()`
+  (`src/main/kotlin/org/octopusden/octopus/automation/teamcity/TeamcityCreateBuildChainCommand.kt`)
+  had this shape before this change:
+
+  ```kotlin
+  val defaultJDKVersion = client.getParameter(ConfigurationType.PROJECT, parentProjectId, "JDK_VERSION")
+  component.buildParameters?.javaVersion?.takeIf { it != defaultJDKVersion }?.let { projectJDKVersion ->
+      setBuildTypeParameter(compileConfig.id, "JDK_VERSION", projectJDKVersion)
+  }
+  ```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)